modules/portaudio/portaudio.c

```diff
@@ -223,10 +223,11 @@ static int src_alloc(struct ausrc_st **stp, const struct ausrc *as,
 	if (!stp || !as || !prm)
 		return EINVAL;
 
-	if (str_isset(device))
-		dev_index = atoi(device);
-	else
-		dev_index = Pa_GetDefaultInputDevice();
+	dev = !str_cmp(device, "default")
+		? mediadev_get_default(&ausrc->dev_list)
+		: mediadev_find(&ausrc->dev_list, device);
```

Comment:
This breaks the current behavior because `mediadev_find()` compares only the device name and ignores the index. There might be already applications that rely on the index.

Reply:
The current behavior is undocumented and inconsistent. No other audio module appears to accept indexes. I only realized that `audio_source` expects an index after loading it in the debugger. What's the point in allowing it in `audio_source portaudio`, but not in `/ausrc`? But the `/ausrc` device names appear to be checked centrally. To allow indexes in `/ausrc` it seems you'd have to allow indexes in all audio modules.